datalake/src/main/java/com/sequenceiq/datalake/entity/SdxCluster.java

```diff
@@ -114,6 +114,9 @@
     @Column(name = "sdx_cluster_service_version")
     private String sdxClusterServiceVersion;
 
+    @Column(nullable = false)
+    private boolean detached;
+
```

Comment:
is it not possible to mark datalake as deleted and if it fails then mark it as available again? in this case we need to add this to every repository query every time

Reply:
I don't think it is the correct thing to mark it as deleted even it is not. Moreover, there will be other complications doing that.
1. There could multiple SDX Custers in a deleted state when you pick it in the recovery it will be tricky.
2.

Reply:
yeah, it makes sense, I just wanted to ask if it is possible to avoid usage of a new boolean field, because of repository calls, but if you think it is the best we can do, I think it is fine